src/app/reporting/Engine.cpp

```diff
@@ -512,11 +513,17 @@ CHIP_ERROR Engine::BuildAndSendSingleReportData(ReadHandler * apReadHandler)
 
     VerifyOrExit(apReadHandler != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrExit(apReadHandler->GetSession() != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    // Depending on whether the session supports large payload or not, the
+    // appropriate  max size would be returned for the Report buffer.
```

Comment:
Why do we care about that here?  We ask the ReadHandler for the size to use, it gives us a size, we use it.

On the other hand, the comment on line 535 (after your changes; 528 before your changes) needs fixing to probably talk about fitting in the size the ReadHandler returned or something.

Reply:
Felt that I needed to add a comment above this call but I guess these details are unnecessary here. Will fix the other comment as well.